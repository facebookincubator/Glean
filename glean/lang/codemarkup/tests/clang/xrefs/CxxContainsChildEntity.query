query: codemarkup.cxx.CxxContainsChildEntity _
transform: [gensort, []]
