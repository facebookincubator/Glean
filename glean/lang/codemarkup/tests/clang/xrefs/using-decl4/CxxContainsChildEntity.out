[
  "@generated",
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "M1" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 35,
                "columnBegin": 1,
                "lineEnd": 35,
                "columnEnd": 31
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "ZZZ" },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "M1" } } }
                  }
                }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 35,
                "columnBegin": 16,
                "lineEnd": 35,
                "columnEnd": 28
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N1" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 9,
                "columnBegin": 1,
                "lineEnd": 33,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "namespace_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "N2" },
                  "parent": { "key": { "name": { "key": "N1" } } }
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 11,
                "columnBegin": 1,
                "lineEnd": 29,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "N2" },
                  "parent": { "key": { "name": { "key": "N1" } } }
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 11,
                "columnBegin": 1,
                "lineEnd": 29,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "standalone" } } },
                  "scope": {
                    "namespace_": {
                      "key": {
                        "name": { "key": "N2" },
                        "parent": { "key": { "name": { "key": "N1" } } }
                      }
                    }
                  }
                }
              },
              "signature": {
                "key": {
                  "returns": { "key": "void" },
                  "parameters": [
                    { "name": { "key": "" }, "type": { "key": "Foo::Bar" } },
                    {
                      "name": { "key": "" },
                      "type": { "key": "N1::N2::N3::YYY" }
                    }
                  ]
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 27,
                "columnBegin": 1,
                "lineEnd": 27,
                "columnEnd": 30
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "N2" },
                  "parent": { "key": { "name": { "key": "N1" } } }
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 11,
                "columnBegin": 1,
                "lineEnd": 29,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "standalone" } } },
                  "scope": {
                    "namespace_": {
                      "key": {
                        "name": { "key": "N2" },
                        "parent": { "key": { "name": { "key": "N1" } } }
                      }
                    }
                  }
                }
              },
              "signature": {
                "key": {
                  "returns": { "key": "void" },
                  "parameters": [
                    { "name": { "key": "" }, "type": { "key": "Foo::Bar" } },
                    {
                      "name": { "key": "" },
                      "type": { "key": "N1::N2::N3::YYY" }
                    }
                  ]
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 55,
                "columnBegin": 1,
                "lineEnd": 60,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "N2" },
                  "parent": { "key": { "name": { "key": "N1" } } }
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 11,
                "columnBegin": 1,
                "lineEnd": 29,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "namespace_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "N3" },
                  "parent": {
                    "key": {
                      "name": { "key": "N2" },
                      "parent": { "key": { "name": { "key": "N1" } } }
                    }
                  }
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 15,
                "columnBegin": 1,
                "lineEnd": 15,
                "columnEnd": 31
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "N2" },
                  "parent": { "key": { "name": { "key": "N1" } } }
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 11,
                "columnBegin": 1,
                "lineEnd": 29,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "namespace_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "N4" },
                  "parent": {
                    "key": {
                      "name": { "key": "N2" },
                      "parent": { "key": { "name": { "key": "N1" } } }
                    }
                  }
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 16,
                "columnBegin": 1,
                "lineEnd": 16,
                "columnEnd": 31
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "N2" },
                  "parent": { "key": { "name": { "key": "N1" } } }
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 11,
                "columnBegin": 1,
                "lineEnd": 29,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "Foo" },
                  "scope": {
                    "namespace_": {
                      "key": {
                        "name": { "key": "N2" },
                        "parent": { "key": { "name": { "key": "N1" } } }
                      }
                    }
                  }
                }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 21,
                "columnBegin": 1,
                "lineEnd": 25,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "N2" },
                  "parent": { "key": { "name": { "key": "N1" } } }
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 11,
                "columnBegin": 1,
                "lineEnd": 29,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "XXX" },
                  "scope": {
                    "namespace_": {
                      "key": {
                        "name": { "key": "N2" },
                        "parent": { "key": { "name": { "key": "N1" } } }
                      }
                    }
                  }
                }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 13,
                "columnBegin": 1,
                "lineEnd": 13,
                "columnEnd": 13
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "N3" },
                  "parent": {
                    "key": {
                      "name": { "key": "N2" },
                      "parent": { "key": { "name": { "key": "N1" } } }
                    }
                  }
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 15,
                "columnBegin": 1,
                "lineEnd": 15,
                "columnEnd": 31
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "YYY" },
                  "scope": {
                    "namespace_": {
                      "key": {
                        "name": { "key": "N3" },
                        "parent": {
                          "key": {
                            "name": { "key": "N2" },
                            "parent": { "key": { "name": { "key": "N1" } } }
                          }
                        }
                      }
                    }
                  }
                }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 15,
                "columnBegin": 16,
                "lineEnd": 15,
                "columnEnd": 28
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "N4" },
                  "parent": {
                    "key": {
                      "name": { "key": "N2" },
                      "parent": { "key": { "name": { "key": "N1" } } }
                    }
                  }
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 16,
                "columnBegin": 1,
                "lineEnd": 16,
                "columnEnd": 31
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "callme" } } },
                  "scope": {
                    "namespace_": {
                      "key": {
                        "name": { "key": "N4" },
                        "parent": {
                          "key": {
                            "name": { "key": "N2" },
                            "parent": { "key": { "name": { "key": "N1" } } }
                          }
                        }
                      }
                    }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 16,
                "columnBegin": 16,
                "lineEnd": 16,
                "columnEnd": 28
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "Foo" },
                  "scope": {
                    "namespace_": {
                      "key": {
                        "name": { "key": "N2" },
                        "parent": { "key": { "name": { "key": "N1" } } }
                      }
                    }
                  }
                }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 21,
                "columnBegin": 1,
                "lineEnd": 25,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "bar" } } },
                  "scope": {
                    "recordWithAccess": {
                      "record": {
                        "key": {
                          "name": { "key": "Foo" },
                          "scope": {
                            "namespace_": {
                              "key": {
                                "name": { "key": "N2" },
                                "parent": { "key": { "name": { "key": "N1" } } }
                              }
                            }
                          }
                        }
                      },
                      "access": 0
                    }
                  }
                }
              },
              "signature": {
                "key": {
                  "returns": { "key": "void" },
                  "parameters": [
                    { "name": { "key": "" }, "type": { "key": "N1::N2::XXX" } },
                    {
                      "name": { "key": "" },
                      "type": { "key": "N1::N2::N3::YYY" }
                    }
                  ]
                }
              },
              "method": {
                "isVirtual": false,
                "isConst": false,
                "isVolatile": false,
                "refQualifier": 0
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 24,
                "columnBegin": 3,
                "lineEnd": 24,
                "columnEnd": 20
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "Foo" },
                  "scope": {
                    "namespace_": {
                      "key": {
                        "name": { "key": "N2" },
                        "parent": { "key": { "name": { "key": "N1" } } }
                      }
                    }
                  }
                }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 21,
                "columnBegin": 1,
                "lineEnd": 25,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "foo" } } },
                  "scope": {
                    "recordWithAccess": {
                      "record": {
                        "key": {
                          "name": { "key": "Foo" },
                          "scope": {
                            "namespace_": {
                              "key": {
                                "name": { "key": "N2" },
                                "parent": { "key": { "name": { "key": "N1" } } }
                              }
                            }
                          }
                        }
                      },
                      "access": 0
                    }
                  }
                }
              },
              "signature": {
                "key": {
                  "returns": { "key": "N1::N2::Foo::Bar" },
                  "parameters": [
                    { "name": { "key": "" }, "type": { "key": "N1::N2::Foo" } },
                    {
                      "name": { "key": "" },
                      "type": { "key": "N1::N2::Foo::Bar" }
                    }
                  ]
                }
              },
              "method": {
                "isVirtual": false,
                "isConst": false,
                "isVolatile": false,
                "refQualifier": 0
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 23,
                "columnBegin": 3,
                "lineEnd": 23,
                "columnEnd": 19
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "Foo" },
                  "scope": {
                    "namespace_": {
                      "key": {
                        "name": { "key": "N2" },
                        "parent": { "key": { "name": { "key": "N1" } } }
                      }
                    }
                  }
                }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 21,
                "columnBegin": 1,
                "lineEnd": 25,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "Bar" },
                  "scope": {
                    "recordWithAccess": {
                      "record": {
                        "key": {
                          "name": { "key": "Foo" },
                          "scope": {
                            "namespace_": {
                              "key": {
                                "name": { "key": "N2" },
                                "parent": { "key": { "name": { "key": "N1" } } }
                              }
                            }
                          }
                        }
                      },
                      "access": 0
                    }
                  }
                }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 22,
                "columnBegin": 3,
                "lineEnd": 22,
                "columnEnd": 15
              }
            }
          }
        }
      }
    }
  }
]